### ISSUE TITLE
About cls_criterion() and pt

### ISSUE THREAD

**Issue by XiGuan6** [ailearn-ml/RDA#2]
First, in resample_loss.py on line 86,
`logpt = self.cls_criterion(cls_score.clone(), label, weight=None, reduction='none', avg_factor=avg_factor)`
`pt = torch.exp(-logpt)`
`wtloss = self.cls_criterion(cls_score, label.float(), weight=weight, reduction='none')`
`loss = alpha_t * ((1 - pt) ** self.gamma) * wtloss`
`alpha_t = torch.where(label == 1, self.alpha, 1 - self.alpha)`
`loss = alpha_t * ((1 - pt) ** self.gamma) * wtloss`

The pt calculated here using cls_criterion() and torch.exp() is not the f(x) in the paper. 
I look at the code in https://github.com/wutong16/DistributionBalancedLoss, in multi-label classification, it is correct, but in the label distribution learning, there will be a mistake.

Second, in resample_loss.py on line 83,
`cls_score, weight = self.logit_reg_functions(label.float(), cls_score, weight)`
logit was adjusted using logit_reg_functions() for class-specific bias,  Why subtract a bias value in cls_criterion()
`pred = torch.exp(pred - torch.max(pred, dim=1, keepdim=True)[0])`
`pred = pred / torch.sum(pred, dim=1, keepdim=True)`

**Comment by ailearn-ml** [comment]
Dear XiGuan6,

Thank you for your attention to our work. 

For the first point, we have updated the code to the latest version where the pt calculated correctly. Thank you very much!

For the second point, since the logit has been adjusted using logit_reg_functions(), we can now use it to form a distribution. To prevent the effect of possibly too large logit on the numerical calculation, we subtract the largest logit corresponding to each sample before calculating the distribution, which does not affect the resulting distribution. Specific information can be referred to https://stats.stackexchange.com/questions/338285/how-does-the-subtraction-of-the-logit-maximum-improve-learning.

**Comment by XiGuan6** [comment]
Thank you very much
